subsys/net/ip/net_context.c

```diff
@@ -825,10 +825,19 @@ NET_CONN_CB(tcp_established)
 	}
 
 	set_appdata_values(pkt, IPPROTO_TCP);
-	context->tcp->send_ack += net_pkt_appdatalen(pkt);
+
+	uint16_t data_len = net_pkt_appdatalen(pkt);
+	if (data_len > get_recv_wnd(context->tcp)) {
+		NET_ERR("Context %p: overflow of recv window (%d vs %d), pkt dropped",
+			context, get_recv_wnd(context->tcp), data_len);
+		return NET_DROP;
```

Comment:
This seems like the wrong behavior.  I mean,  yeah, it's a stack bug on the other side and they blew the limit you set.  Nonetheless, they transmitted the data and we received it without overflow.   That's not an error condition, we should present the data to the app and attempt to continue.

I mean, just to geek out: I went to get a quote for the "be conservative in what you emit but liberal in what you accept" maxim and was reminded by wikipedia that it's **literally from the TCP RFC** (thus "Postel's Law"):

TCP implementations should follow a general principle of robustness: be conservative in what you do, be liberal in what you accept from others.